### DISCUSSION TITLE
vim-multiple-cursors yank

### DISCUSSION THREAD

**Discussion by jacek-kurlit** [JetBrains/ideavim#587] [Q&A] [answered]
Hi,

I'm not sure if this is bug or limitation that this plugin originally had but my issue is - when I select few occurrences(multiple carets) and yank them, only text under the last cursor is put to registers. As a result, when pasting text (still while having multiple cursors) only last text is placed .

What I would expect is behavior same as intellij or https://github.com/mg979/vim-visual-multi
Text under each cursor should be remembered and when pasting it, each cursor should have corresponding text.

Is this possible or it requires to implement new approach to achieve this?

**Comment by AlexPl292** [comment, marked as answer]
Hi, could you please try this in the last version of IdeaVim (2.1.0)?

**Reply by jacek-kurlit** [reply]
Hi,
I have updated plugin to 2.1.0 but it didn't help.

**Reply by lippfi** [reply]
Hi! Could you please tell me your clipboard opion's value? ( `set clipboard?`  )

**Reply by jacek-kurlit** [reply]
set clipboard+=unnamed

**Reply by lippfi** [reply]
That's the reason
We just recently supported individual registers for each caret and have not yet polished the behavior.
The fact of the matter is that unnamed register is unique (system clipboard), so all the carets forced to use system register instead of their own. For now you can specify register excplicitly e.g. `""y` to yank to the `"` register) or use default clipboard option value.
In future releases we will fix it and ignore `clipboard` option when multiple carets are used.
Thank you for your report!

**Reply by jacek-kurlit** [reply]
Indeed, using default clipboard helps. Thank you!
